projects/components/src/datagrid/datagrid.component.ts

```diff
@@ -172,6 +182,24 @@ export interface PaginationConfiguration {
     shouldShowPageNumberInput?: boolean;
 }
 
+/**
+ * The configuration object that is passed to the detail row component.
+ */
+export interface DetailRowConfig<R> {
```

Comment:
I don't think this will be required with the above suggested approach.

If needed, we can make any XyzRowDetailComponent implement ComponentRenderer interface and pass the restItem as it's config

Reply:
I still needed to provide this interface to specify to the client what config object we will provide them in their details component. Because the component renderer outlet provides a single config argument this felt necessary. Let me know if you still think it can be removed, would also like to remove it of possible.